ssh-notes/ssh-connection-problems.md

```diff
@@ -0,0 +1,138 @@
+  ```
+  1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN mode DEFAULT group default qlen 1000
+      link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
+  2: bond0: <BROADCAST,MULTICAST,MASTER> mtu 1500 qdisc noop state DOWN mode DEFAULT group default qlen 1000
+      link/ether 8e:d4:8b:55:37:c8 brd ff:ff:ff:ff:ff:ff
+  3: dummy0: <BROADCAST,NOARP> mtu 1500 qdisc noop state DOWN mode DEFAULT group default qlen 1000
+      link/ether 06:63:1c:4d:3b:48 brd ff:ff:ff:ff:ff:ff
+  4: eth0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc mq state UP mode DEFAULT group default qlen 1000
+      link/ether 00:15:5d:2e:42:5f brd ff:ff:ff:ff:ff:ff
+  5: tunl0@NONE: <NOARP> mtu 1480 qdisc noop state DOWN mode DEFAULT group default qlen 1000
+      link/ipip 0.0.0.0 brd 0.0.0.0
+  6: sit0@NONE: <NOARP> mtu 1480 qdisc noop state DOWN mode DEFAULT group default qlen 1000
+      link/sit 0.0.0.0 brd 0.0.0.0
+  ```
+7. Locate the 'eth0' entry and note the MTU value.
+
+So once, you have done all this you should have the following information
+
+- For the network interface(s) that your laptop uses to connect to the internet
+  - An IDX value
+  - An MTU value
+
+- For the network interface the WSL machine uses (there will only be one of these)
+  - An MTU value
+
+If the MTU values for the Windows environment does NOT match the one that the WSL machine sees, then you have found the cause of the SSH hanging problem.
+
+## How to Fix the Problem
+
+We need to make the MTU in Windows match the MTU in the WSL machine (or vice versa).  So to correct the situation you must do one of two things - either modify the MTU value in Windows, or modify the MTU value in the WSL environment.
+
+#### In Windows
+You need to start a Windows command prompt (or Powershell).  The command prompt/power shell needs to be running as the Administrator.  Then you must run the following command:
+
+```text
+netsh interface ipv4 set subinterface XXX mtu=YYY store=persistent
+```
+
+Where you replace
+
+- 'XXX' with index you noted down 'Preparation' section.
+- 'YYY' with the MTU value from the WSL machine
+
+#### In WSL
+You need to start a WSL Shell and then run the following commands:
+
+```text
+sudo apt install net-tools
+sudo ifconfig eth0 mtu YYY
+```
+
+Where you replace
+
+- 'YYY' with the MTU value from the Windows machine
+
+## Conclusion
+
+This will fix the problem with SSH hanging.  You can test that the fix has worked by running the command
+
+```text
+ssh -T git@github.com
+```
+
+# PLEASE NOTE
+The two methods outlined above will **not** remember the MTU changes if you reboot the machine.
+
+In the WSL machine to persist these changes you can either add the ifconfig command to your bashrc file.
+
+Under Windows you may need to set up a command file that you run after you reboot the machine.  There is an alternative, but this requires editing the registry and is quite complicated.  See the file
```

Comment:
... See the file ???

Docs should be linked together for ease of navigation.

Reply:
Done.  Brain freeze - forgot to add file name